### PR TITLE
refactor: docker getRegistryRepository

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Renames results fields in `docker` datasource `getRegistryRepository` to make an existing bug more clear.

## Context:

#10135

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/docker/__snapshots__/common.spec.ts.snap [hunk 1]
```diff
@@ -2,28 +2,28 @@
...
+  "registryHost": "https://registry:5000",
 }
 `;
 
 exports[`datasource/docker/common getRegistryRepository supports http registryUrls 1`] = `
 Object {
-  "registry": "http://my.local.registry/prefix",
-  "repository": "image",
+  "dockerRepository": "image",
+  "registryHost": "http://my.local.registry/prefix",
```
Comment (rarkins) [inline lib/datasource/docker/__snapshots__/common.spec.ts.snap:13]: Wrong

File: lib/datasource/docker/__snapshots__/common.spec.ts.snap [hunk 2]
```diff
@@ -2,28 +2,28 @@
...
 }
 `;
 
 exports[`datasource/docker/common getRegistryRepository supports registryUrls 1`] = `
 Object {
-  "registry": "https://my.local.registry/prefix",
-  "repository": "image",
+  "dockerRepository": "image",
+  "registryHost": "https://my.local.registry/prefix",
```
Comment (rarkins) [inline lib/datasource/docker/__snapshots__/common.spec.ts.snap:20]: Wrong

File: lib/datasource/docker/__snapshots__/common.spec.ts.snap [hunk 3]
```diff
@@ -2,28 +2,28 @@
...
 }
 `;
 
 exports[`datasource/docker/common getRegistryRepository supports schemeless registryUrls 1`] = `
 Object {
-  "registry": "https://my.local.registry/prefix",
-  "repository": "image",
+  "dockerRepository": "image",
+  "registryHost": "https://my.local.registry/prefix",
```
Comment (rarkins) [inline lib/datasource/docker/__snapshots__/common.spec.ts.snap:27]: Wrong

File: lib/datasource/docker/common.ts [hunk 4]
```diff
@@ -47,11 +47,11 @@ async function getECRAuthToken(
 }
 
 export async function getAuthHeaders(
-  registry: string,
+  registryHost: string,
   dockerRepository: string
 ): Promise<OutgoingHttpHeaders | null> {
   try {
-    const apiCheckUrl = `${registry}/v2/`;
+    const apiCheckUrl = `${registryHost}/v2/`;
```
Comment (rarkins) [inline lib/datasource/docker/common.ts:54]: This will fail in any of the "Wrong" scenarios above

File: lib/datasource/docker/common.ts [hunk 5]
```diff
@@ -214,20 +214,22 @@ export function extractDigestFromResponse(
...
-    const headers = await getAuthHeaders(registry, dockerRepository);
+    const headers = await getAuthHeaders(registryHost, dockerRepository);
     if (!headers) {
       logger.debug('No docker auth found - returning');
       return null;
     }
     headers.accept =
       'application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.docker.distribution.manifest.v2+json';
-    const url = `${registry}/v2/${dockerRepository}/manifests/${tag}`;
+    const url = `${registryHost}/v2/${dockerRepository}/manifests/${tag}`;
```
Comment (rarkins) [inline lib/datasource/docker/common.ts:232]: This will fail in any of the "Wrong" scenarios above

File: lib/datasource/docker/common.ts [hunk 6]
```diff
@@ -358,18 +360,22 @@ export async function getLabels(
...
 
-    const headers = await getAuthHeaders(registry, dockerRepository);
+    const headers = await getAuthHeaders(registryHost, dockerRepository);
     // istanbul ignore if: Should never be happen
     if (!headers) {
       logger.debug('No docker auth found - returning');
       return {};
     }
-    const url = `${registry}/v2/${dockerRepository}/blobs/${configDigest}`;
+    const url = `${registryHost}/v2/${dockerRepository}/blobs/${configDigest}`;
```
Comment (rarkins) [inline lib/datasource/docker/common.ts:378]: This will fail in any of the "Wrong" scenarios above

File: lib/datasource/docker/index.ts [hunk 7]
```diff
@@ -71,9 +71,9 @@ async function getTags(
     }
     // AWS ECR limits the maximum number of results to 1000
     // See https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_DescribeRepositories.html#ECR-DescribeRepositories-request-maxResults
-    const limit = ecrRegex.test(registry) ? 1000 : 10000;
-    let url = `${registry}/v2/${repository}/tags/list?n=${limit}`;
-    const headers = await getAuthHeaders(registry, repository);
+    const limit = ecrRegex.test(registryHost) ? 1000 : 10000;
+    let url = `${registryHost}/v2/${repository}/tags/list?n=${limit}`;
```
Comment (rarkins) [inline lib/datasource/docker/index.ts:75]: This will fail in any of the "Wrong" scenarios above

**Review by viceice — COMMENTED** [review]
needs deconflicting now 🙃

**Pull request merged**